cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/ProvisionerExtensionLoader.java

```diff
@@ -42,7 +43,11 @@
 
   private static Set<String> createAllowedResources() {
     try {
-      return ClassPathResources.getResourcesWithDependencies(Provisioner.class.getClassLoader(), Provisioner.class);
+      Set<String> resources = new HashSet<>();
+      ClassLoader provisionerClassLoader = Provisioner.class.getClassLoader();
+      resources.addAll(ClassPathResources.getResourcesWithDependencies(provisionerClassLoader, Provisioner.class));
+      resources.addAll(ClassPathResources.getResourcesWithDependencies(provisionerClassLoader, MessagingService.class));
```

Comment:
would this be loaded for all provisioners?

Reply:
Yes, the provisioners filter against the same set of allowed resources/packages